PowerEditor/src/TinyXml/tinyxml.cpp

```diff
@@ -567,19 +567,19 @@ void TiXmlElement::Print( Win32_IO_File& cfile, int depth ) const
 	int i;
 	for ( i=0; i<depth; i++ )
 	{
-		cfile.write(four_ws, 4);
+		cfile.writeStr(four_ws);
 		//generic_fprintf( cfile, TEXT("    ") );
 	}
 
-	generic_string tagOpenValue = tagOpenSymb + value;
-	std::string tagOpenValueA = wstring2string(tagOpenValue, CP_UTF8);
-	cfile.write(tagOpenValueA.c_str(), static_cast<unsigned long>(tagOpenValueA.length()));
+	std::string tagOpenValue = tagOpenSymb;
+	tagOpenValue += wstring2string(value, CP_UTF8);
+	cfile.writeStr(tagOpenValue.c_str());
```

Comment:
Look my suggestion above.
In case you implement `write(const std::string& str)` method you'll simply do `cfile.writeStr(tagOpenValue);` here and inside `tagOpenValue.length()` will be used instead of `strlen()`. Thus no time will be wasted to loop through all the string chars.
This is valid for all `std::string` `write()` invocations below as well.